### PR TITLE
UI panels initial commit

### PR DESCRIPTION
**Pull request by jdent02**

### CODE REVIEW

File: properties/scene.py [hunk 1]
```diff
@@ -48,24 +54,173 @@ class RendermanSceneSettings(RendermanBasePropertyGroup):
...
+        name="X",
+        description="X coordinate of bucket spiral start",
+        min=-1, default=-1)
+
+    bucket_sprial_y = IntProperty(
+        name="Y",
+        description="Y coordinate of bucket spiral start",
+        min=-1, default=-1)
+
+    shadingrate = FloatProperty(
```
Comment (bsavery) [inline properties/scene.py]: We should probably rename this property to micropolygon length

File: properties/scene.py [hunk 2]
```diff
@@ -48,24 +54,173 @@ class RendermanSceneSettings(RendermanBasePropertyGroup):
...
+        name="Rendering Threads",
+        description="Number of processor threads to use.  Note, 0 uses all cores, -1 uses all cores but one.",
+        min=-32, max=32, default=-1)
+
+    use_statistics = BoolProperty(
+        name="Statistics",
+        description="Print statistics to stats.xml after render",
+        default=False)
+
+    editor_override = StringProperty(
```
Comment (bsavery) [inline properties/scene.py]: No one uses this... might as well remove.

File: ui/__init__.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import os
+
+# get all .py files in directory
+__all__ = [basename(f)[:-3] for f in glob.glob(dirname(__file__)+"/*.py") \
+           if isfile(f) and basename(f) != '__init__.py']
+
+# try and reload modules if already loaded
+import importlib as imp
+for module in __all__:
+    imp.import_module('.' + module, package=__name__)
```
Comment (bsavery) [inline ui/__init__.py]: Python files should always have a newline at the end.

File: ui/base_classes.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        op = col.operator(operator, icon="ZOOMOUT", text="")
+        op.context = opcontext
+        op.collection = prop_coll
+        op.collection_index = collection_index
+        op.action = 'REMOVE'
+
+        if hasattr(ptr, prop_coll) and len(getattr(ptr, prop_coll)) > 0 and \
+                getattr(ptr, collection_index) >= 0:
+            item = getattr(ptr, prop_coll)[getattr(ptr, collection_index)]
+            self.draw_item(layout, context, item)
```
Comment (bsavery) [inline ui/base_classes.py]: What do you think about passing the draw_item method in to the _draw_collection method.  I.e. remove the draw_item method as a class method, but make it a parameter of this function.

File: ui/advanced_settings.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        col = layout.column()
+        col.label("Pixel Filter:")
+        col.prop(rm, "pixelfilter")
+        row = col.row(align=True)
+        row.prop(rm, "pixelfilter_x", text="Size X")
+        row.prop(rm, "pixelfilter_y", text="Size Y")
+
+        # layout.separator()
+        # col = layout.column()
+        # col.prop(rm, "dark_falloff")
```
Comment (bsavery) [inline ui/advanced_settings.py]: Why is darkfalloff commented out?

**Comment by bsavery** [comment]
Not too bad.  Just a few minor things.  OH and also you should probably make the names of the files advanced_scene_panel, scene_motion_blur, etc.

File: properties/scene.py (hunk 1 above)
Comment (jdent02) [inline properties/scene.py]: Okay.

File: ui/base_classes.py (hunk 4 above)
Comment (jdent02) [inline ui/base_classes.py]: I see what you're saying but I'm not sure how to code it properly given that the item parameters are going to change for every ui panel it appears in.

File: ui/advanced_settings.py (hunk 5 above)
Comment (jdent02) [inline ui/advanced_settings.py]: Was thinking of moving it to the sampling panel

File: ui/base_classes.py (hunk 4 above)
Comment (jdent02) [inline ui/base_classes.py]: Or maybe I'm misunderstanding what you're saying.  I took the draw_item method out of the class definition.

File: ui/advanced_settings.py (hunk 5 above)
Comment (bsavery) [inline ui/advanced_settings.py]: Ok sounds good.

**Pull request merged**